core/src/main/scala/filodb.core/metadata/Dataset.scala

```diff
@@ -88,6 +89,8 @@ object Dataset {
                                     asInstanceOf[SortKeyHelper[K]]
       case java.lang.Double.TYPE => (new DoubleKeyHelper(options.segmentSize.toDouble)).
                                     asInstanceOf[SortKeyHelper[K]]
+      case StringClass         => (new StringKeyHelper(options.segmentSize.toInt)).
+                                    asInstanceOf[SortKeyHelper[K]]
```

Comment:
We shouldn't, because it would be outside the defined types, and in this case MatchError would probably be right.